charmcraft/commands/build.py

```diff
@@ -161,13 +143,44 @@ def build_charm(self, bases_config: BasesConfiguration) -> str:
         """
         logger.debug("Building charm in %r", str(self.buildpath))
 
-        if self.buildpath.exists():
-            shutil.rmtree(str(self.buildpath))
-        self.buildpath.mkdir()
-
-        linked_entrypoint = self.handle_generic_paths()
-        self.handle_dispatcher(linked_entrypoint)
-        self.handle_dependencies()
+        build_env = dict(LANG="C.UTF-8", LC_ALL="C.UTF-8")
+        for key in [
+            "PATH",
```

Comment:
This could use a little more test coverage to validate passthrough env unless I missed it.

A short code comment would be handy too to explain why. 

Please add `no_proxy` to this list as well :D

Reply:
> Also, I assume that the tests were minimally touched, just adjusted for the splitting and no behaviour change. Is that right?

Tests were changed basically in the original `Builder` invocation which was replaced with the corresponding `CharmBuilder` invocation, and package names in mock patching.